File: database/database-annotation-processor/src/test/java/io/koraframework/database/common/annotation/processor/cassandra/AbstractCassandraRepositoryTest.java
package io.koraframework.database.common.annotation.processor.cassandra;

import io.koraframework.database.common.annotation.processor.AbstractRepositoryTest;
import org.intellij.lang.annotations.Language;

import java.util.List;

public abstract class AbstractCassandraRepositoryTest extends AbstractRepositoryTest {
    protected MockCassandraExecutor executor = new MockCassandraExecutor();

    @Override
    protected String commonImports() {
        return super.commonImports() + """
            import io.koraframework.database.cassandra.*;
            import io.koraframework.database.cassandra.mapper.result.*;
            import io.koraframework.database.cassandra.mapper.parameter.*;

            import java.util.concurrent.CompletionStage;

            import com.datastax.oss.driver.api.core.cql.*;
            import com.datastax.oss.driver.api.core.data.*;
            """;
    }

    protected TestObject compileCassandra(List<?> arguments, @Language("java") String... sources) {
        return this.compile(this.executor, arguments, sources);
    }
}

File: database/database-symbol-processor/src/test/kotlin/io/koraframework/database/symbol/processor/cassandra/AbstractCassandraRepositoryTest.kt
package io.koraframework.database.symbol.processor.cassandra

import io.koraframework.database.symbol.processor.AbstractRepositoryTest
import org.intellij.lang.annotations.Language

abstract class AbstractCassandraRepositoryTest : AbstractRepositoryTest() {
    val executor = MockCassandraExecutor()

    override fun commonImports(): String {
        return super.commonImports() + """
            import io.koraframework.database.cassandra.*;
            import io.koraframework.database.cassandra.annotation.*;
            import io.koraframework.database.cassandra.mapper.result.*;
            import io.koraframework.database.cassandra.mapper.parameter.*;

            import java.util.concurrent.CompletionStage;

            import com.datastax.oss.driver.api.core.cql.*;
            import com.datastax.oss.driver.api.core.data.*;
        """.trimIndent()
    }

    protected fun compile(arguments: List<*>, @Language("kotlin") vararg sources: String) = compile(executor, arguments, *sources)
}
